lib/content-types/utils.js

```diff
@@ -0,0 +1,138 @@
+ * @param  {array} config.attributes - attributes config
+ *
+ * @returns {true|string} either true or an error string
+ */
+const attributes = (config) => {
+  if (!config.hasOwnProperty('attributes') || config.attributes === '') {
+    return `Content type '${config.name}' must have attributes`;
+  }
+
+  if (!Array.isArray(config.attributes)) {
+    return `Attributes in content type '${config.name}' must be an array`;
+  }
+
+  if (config.attributes.length < 1) {
+    return `Content type '${config.name}' must have at least one attribute`;
+  }
+
+  // map returns either true or a string; filter leaves us with just an array of strings
+  const attrs = config.attributes.map(attr => {
+    if ((!attr.hasOwnProperty('name') || attr.name === '')) {
+      let ident = attr.type;
+
+      if (attr.hasOwnProperty('id')) {
+        ident = attr.id;
+      }
+
+      return `Attribute '${ident}' in content type '${config.name}' needs a name`;
+    }
+    if (typeof attr.name !== 'string') {
+      return `Attribute names in content type '${config.name}' must be a string`;
+    }
+    if (!attr.id || attr.id === '') {
+      return `Attribute '${attr.name}' in content type '${config.name}' must have an id`;
+    }
+    if (typeof attr.id !== 'string') {
+      return `Attribute id in '${attr.name}' in content type '${config.name}' must be a string`;
+    }
+    if (!attr.type || attr.type === '') {
+      return `Attribute '${attr.name}' in content type '${config.name}' must have a type`;
+    }
+    if (typeof attr.type !== 'string') {
+      return `Attribute type in '${attr.name}' in content type '${config.name}' must be a string.`;
+    }
+
+    return true;
+  }).filter((res) => {
+    if (res !== true) {
+      return true;
+    }
+
+    return false;
+  });
+
+  // no errors are present
+  if (!Array.isArray(attrs) || !attrs.length) {
+    return true;
+  }
+
+  // return the first error
+  return attrs[0];
+};
+
+/**
+ * Check the structure of a content type configuration
+ *
+ * @param {object} config content type configuration
+ *
+ * @returns {true|string} either true or an error string
+ */
+const type = (config) => {
+  if (typeof config !== 'object') {
+    return 'Type must be an object';
+  }
+
+  if (!config.hasOwnProperty('name') || config.name === '') {
+    return 'Content types require a name';
+  }
+
+  if (typeof config.name !== 'string') {
+    return 'Content type name must be string';
+  }
+
+  if (!config.hasOwnProperty('id') || config.id === '') {
+    return 'Content types require an id';
+  }
+
+  if (typeof config.id !== 'string') {
+    return 'Content type id must be string';
+  }
+
+  if (config.id !== slugify(config.id)) {
+    return `${config.id} needs to be written in kebab case (e.g. ${slugify(config.id)})`;
+  }
+
+  const attrs = attributes(config);
+
+  if (attrs !== true) {
+    return attrs;
+  }
+
+  // can't check identifier until we know attributes passes
+  if (config.hasOwnProperty('identifier')) {
```

Comment:
@Snugug this is the common check of `identifier`. It checks everything it could, within the confines of _only_ having the original config for a content-type and not the full merged content-type object.